lms/djangoapps/certificates/views/support.py

```diff
@@ -208,6 +208,17 @@ def regenerate_certificate_for_user(request):
         )
         return HttpResponseServerError(_("An unexpected error occurred while regenerating certificates."))
 
+    try:
```

Comment:
Can you wrap this block into a new function to improve maintainability, documentation and readability?